### PR TITLE
fix(router): accept empty params for router-link

### LINKED ISSUE #1689: Router should support router-link without [router-params]
The following doesn't seem to work with the new router. It never updates the hrefs on the anchors.
```html
<a router-link="sectionA">Section A</a>
<a router-link="sectionB">Section B</a>
<router-outlet></router-outlet>
```

This is because of the isPresent(this._params) check in RouterLink:updateHref
  ```javascript
updateHref() {
  if (isPresent(this._route) && isPresent(this._params)) {
    var newHref = this._router.generate(this._route, this._params);
    DOM.setAttribute(this._domEl, 'href', newHref);
  }
}
```

The problem with this is that this forces you to always define both. However I would want the following unit test to work in outlet_spec.
```javascript
it('should generate link hrefs when there are no router params defined', inject([AsyncTestCompleter], (async) => {
  compile('<a router-link="section A">Section A</a> { <router-outlet></router-outlet> }')
    .then((_) => rtr.config({'path': '/a', 'component': A, 'as': 'sectionA'}))
    .then((_) => rtr.navigate('/a'))
    .then((_) => {
      view.detectChanges();
      expect(view.rootNodes).toHaveText('SectionA { A }');
      expect(DOM.getAttribute(view.rootNodes[0].childNodes[0], 'href')).toEqual('/a');
      async.done();
    });
}));
```

Maybe it would be an option to set params to default to {} in the constructor of RouterLink? The following in RouterLink fixes the previous unit test:
```javascript
constructor(elementRef:ElementRef, router:Router) {
  this._domEl = elementRef.domElement;
  this._router = router;
  this._params = {};
}
```

Although I'm afraid this will generate the newHref and set the href twice if router-params is set which is not ideal.

### PR DESCRIPTION
**Pull request by rkirov**
Closes #1689

### CODE REVIEW

**Comment by btford** [comment]
LGTM

File: modules/angular2/src/router/router_link.js [hunk 1]
```diff
@@ -39,11 +40,15 @@ export class RouterLink {
...
   _router:Router;
-  //TODO: handle click events
 
   constructor(elementRef:ElementRef, router:Router) {
     this._domEl = elementRef.domElement;
     this._router = router;
+    this._params = StringMapWrapper.create();
+    DOM.on(this._domEl, 'click', (evt) => {
+      evt.preventDefault();
+      r.navigate(evt.target.href);
```
Comment (btford) [inline modules/angular2/src/router/router_link.js]: should "r" be this._router ?

**Comment by rkirov** [comment]
take another look @btford

**Comment by btford** [comment]
LGTM

**Pull request closed without merging**